enterprise/server/workflow/workflow.go

```diff
@@ -261,48 +247,59 @@ func (ws *workflowService) readWorkflowForWebhook(ctx context.Context, webhookID
 	return tw, nil
 }
 
-func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tables.Workflow, ci *workflowcmd.CommandInfo, instanceName string) (*repb.Digest, error) {
-	script, err := workflowcmd.GenerateShellScript(ci)
-	if err != nil {
-		return nil, err
-	}
-
+func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tables.Workflow, wd *webhook_data.WebhookData, ak *tables.APIKey, instanceName string) (*repb.Digest, error) {
 	cache := ws.env.GetCache()
 	if cache == nil {
 		return nil, status.UnavailableError("No cache configured.")
 	}
 
-	scriptName := "run.sh"
-	scriptDigest, err := cachetools.UploadBlobToCAS(ctx, cache, instanceName, script)
+	runnerBinName := "buildbuddy_action_runner"
+	runnerBinBytes, err := runnerBinary()
+	if err != nil {
+		return nil, err
+	}
+	runnerBinDigest, err := cachetools.UploadBlobToCAS(ctx, cache, instanceName, runnerBinBytes)
```

Comment:
I think this is fine but it will buffer the whole "runner" blob in memory. Depending on how big it is that could be a bad thing. If you wanted to stream it instead you could:

make uploadBytesToCache public in cachetools.go.
return a io.ReadSeeker from runnerBinary() -- just returning the open file object is enough here
Call uploadBytesToCache(ctx, cache, instanceName, file) -- it will read from the file and write the bytes directly to the cache in chunks, rather than buffering the whole thing.